Settings: Network Traffic Indicator (2/3)

Original patch: https://review.mfunz.com/#/c/13980/

project packages/apps/Settings/
diff --git a/packages/apps/Settings/res/values-ar/cm_strings.xml b/packages/apps/Settings/res/values-ar/cm_strings.xml
index fe33c10a2..5de516135 100644
--- a/packages/apps/Settings/res/values-ar/cm_strings.xml
+++ b/packages/apps/Settings/res/values-ar/cm_strings.xml
@@ -1048,4 +1048,11 @@
   <!-- Format string for fingerprint location message -->
   <!-- Fingerprint sensor locations -->
   <!-- SIM status format string -->
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">سرعة الشبكة</string>
+  <string name="show_network_speed_summary">عرض سرعة الشبكة في شريط الحالة</string>
+  <string name="status_bar_network_traffic_style_disable">معطل</string>
+  <string name="status_bar_network_traffic_style_up">صادرة</string>
+  <string name="status_bar_network_traffic_style_down">الوارده</string>
+  <string name="status_bar_network_traffic_style_all">الصادرة والواردة</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-be/cm_strings.xml b/packages/apps/Settings/res/values-be/cm_strings.xml
index e771cd87a..dbe105671 100644
--- a/packages/apps/Settings/res/values-be/cm_strings.xml
+++ b/packages/apps/Settings/res/values-be/cm_strings.xml
@@ -1015,4 +1015,11 @@
   <!-- Format string for fingerprint location message -->
   <!-- Fingerprint sensor locations -->
   <!-- SIM status format string -->
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Хуткасць сеткі</string>
+  <string name="show_network_speed_summary">Паказваць хуткасць сеткі ў радку стану</string>
+  <string name="status_bar_network_traffic_style_disable">Выключана</string>
+  <string name="status_bar_network_traffic_style_up">Выходныя</string>
+  <string name="status_bar_network_traffic_style_down">Уваходныя</string>
+  <string name="status_bar_network_traffic_style_all">Выходныя i ўваходныя</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-bg/cm_strings.xml b/packages/apps/Settings/res/values-bg/cm_strings.xml
index 3aedb9dea..d4edcbd43 100644
--- a/packages/apps/Settings/res/values-bg/cm_strings.xml
+++ b/packages/apps/Settings/res/values-bg/cm_strings.xml
@@ -1098,4 +1098,11 @@
   <string name="security_settings_fingerprint_sensor_location_right">Дясна страна</string>
   <!-- SIM status format string -->
   <string name="sim_status_format_string">Глас: <xliff:g id="voice_status">%1$s</xliff:g> / Данни: <xliff:g id="data_status">%2$s</xliff:g></string>
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Мрежова скорост</string>
+  <string name="show_network_speed_summary">Показва скорост на мрежа в лентата на състоянието</string>
+  <string name="status_bar_network_traffic_style_disable">Забранено</string>
+  <string name="status_bar_network_traffic_style_up">Изходящ</string>
+  <string name="status_bar_network_traffic_style_down">Входящи</string>
+  <string name="status_bar_network_traffic_style_all">Изходящи и входящи</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-ca/cm_strings.xml b/packages/apps/Settings/res/values-ca/cm_strings.xml
index 6678c7282..f003795b4 100644
--- a/packages/apps/Settings/res/values-ca/cm_strings.xml
+++ b/packages/apps/Settings/res/values-ca/cm_strings.xml
@@ -1098,4 +1098,11 @@
   <string name="security_settings_fingerprint_sensor_location_right">costat dret</string>
   <!-- SIM status format string -->
   <string name="sim_status_format_string">Veu: <xliff:g id="voice_status">%1$s</xliff:g> / Dades: <xliff:g id="data_status">%2$s</xliff:g></string>
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Velocitat de la xarxa</string>
+  <string name="show_network_speed_summary">Mostra la velocitat de la xarxa a la barra d\'estat</string>
+  <string name="status_bar_network_traffic_style_disable">Desactivat</string>
+  <string name="status_bar_network_traffic_style_up">Enviats</string>
+  <string name="status_bar_network_traffic_style_down">Rebuts</string>
+  <string name="status_bar_network_traffic_style_all">Enviats i rebuts</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-cs/cm_strings.xml b/packages/apps/Settings/res/values-cs/cm_strings.xml
index d22152601..917d43c18 100644
--- a/packages/apps/Settings/res/values-cs/cm_strings.xml
+++ b/packages/apps/Settings/res/values-cs/cm_strings.xml
@@ -1096,4 +1096,11 @@
   <string name="security_settings_fingerprint_sensor_location_right">pravá strana</string>
   <!-- SIM status format string -->
   <string name="sim_status_format_string">Hlas: <xliff:g id="voice_status">%1$s</xliff:g> / Data: <xliff:g id="data_status">%2$s</xliff:g></string>
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Rychlost sítě</string>
+  <string name="show_network_speed_summary">Zobrazit rychlost sítě ve stavové liště</string>
+  <string name="status_bar_network_traffic_style_disable">Nezobrazovat</string>
+  <string name="status_bar_network_traffic_style_up">Odchozí</string>
+  <string name="status_bar_network_traffic_style_down">Příchozí</string>
+  <string name="status_bar_network_traffic_style_all">Odchozí a příchozí</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-da/cm_strings.xml b/packages/apps/Settings/res/values-da/cm_strings.xml
index 2181214c0..060d8424d 100644
--- a/packages/apps/Settings/res/values-da/cm_strings.xml
+++ b/packages/apps/Settings/res/values-da/cm_strings.xml
@@ -1058,4 +1058,11 @@
   <string name="security_settings_fingerprint_sensor_location_right">højre side</string>
   <!-- SIM status format string -->
   <string name="sim_status_format_string">Stemme: <xliff:g id="voice_state">%1$s</xliff:g>, data: <xliff:g id="data_state">%2$s</xliff:g></string>
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Netværkshastighed</string>
+  <string name="show_network_speed_summary">Vis netværkshastighed på statusbjælken</string>
+  <string name="status_bar_network_traffic_style_disable">Deaktiveret</string>
+  <string name="status_bar_network_traffic_style_up">Udgående</string>
+  <string name="status_bar_network_traffic_style_down">Indgående</string>
+  <string name="status_bar_network_traffic_style_all">Udgående og indgående</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-de/cm_strings.xml b/packages/apps/Settings/res/values-de/cm_strings.xml
index 102f48109..0d844ee5a 100644
--- a/packages/apps/Settings/res/values-de/cm_strings.xml
+++ b/packages/apps/Settings/res/values-de/cm_strings.xml
@@ -1098,4 +1098,11 @@
   <string name="security_settings_fingerprint_sensor_location_right">rechten Seite</string>
   <!-- SIM status format string -->
   <string name="sim_status_format_string">Anrufe: <xliff:g id="voice_state">%1$s</xliff:g> / Daten: <xliff:g id="data_state">%2$s</xliff:g></string>
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Netzwerkgeschwindigkeit</string>
+  <string name="show_network_speed_summary">Netzwerkgeschwindigkeit in Statusleiste anzeigen</string>
+  <string name="status_bar_network_traffic_style_disable">Deaktiviert</string>
+  <string name="status_bar_network_traffic_style_up">Ausgehend</string>
+  <string name="status_bar_network_traffic_style_down">Eingehend</string>
+  <string name="status_bar_network_traffic_style_all">Ausgehend und Eingehend</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-el/cm_strings.xml b/packages/apps/Settings/res/values-el/cm_strings.xml
index aa38529e9..e0a52b458 100644
--- a/packages/apps/Settings/res/values-el/cm_strings.xml
+++ b/packages/apps/Settings/res/values-el/cm_strings.xml
@@ -1095,4 +1095,11 @@
   <string name="security_settings_fingerprint_sensor_location_right">στη δεξιά πλευρά</string>
   <!-- SIM status format string -->
   <string name="sim_status_format_string">Φωνή: <xliff:g id="voice_status">%1$s</xliff:g> / Δεδομένα: <xliff:g id="data_status">%2$s</xliff:g></string>
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Ταχύτητα δικτύου</string>
+  <string name="show_network_speed_summary">Εμφάνιση ταχύτητας δικτύου στη γραμμή κατάστασης</string>
+  <string name="status_bar_network_traffic_style_disable">Απενεργοποιημένο</string>
+  <string name="status_bar_network_traffic_style_up">Εξερχομενα</string>
+  <string name="status_bar_network_traffic_style_down">Εισερχόμενα</string>
+  <string name="status_bar_network_traffic_style_all">Εξερχόμενων και εισερχόμενων</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-en-rAU/cm_strings.xml b/packages/apps/Settings/res/values-en-rAU/cm_strings.xml
index dddae3fb4..f9e8a0eda 100644
--- a/packages/apps/Settings/res/values-en-rAU/cm_strings.xml
+++ b/packages/apps/Settings/res/values-en-rAU/cm_strings.xml
@@ -772,4 +772,11 @@
   <!-- Format string for fingerprint location message -->
   <!-- Fingerprint sensor locations -->
   <!-- SIM status format string -->
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Network speed</string>
+  <string name="show_network_speed_summary">Show network speed in status bar</string>
+  <string name="status_bar_network_traffic_style_disable">Disabled</string>
+  <string name="status_bar_network_traffic_style_up">Outgoing</string>
+  <string name="status_bar_network_traffic_style_down">Incoming</string>
+  <string name="status_bar_network_traffic_style_all">Outgoing and incoming</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-en-rGB/cm_strings.xml b/packages/apps/Settings/res/values-en-rGB/cm_strings.xml
index b76ef67fd..1a2698eae 100644
--- a/packages/apps/Settings/res/values-en-rGB/cm_strings.xml
+++ b/packages/apps/Settings/res/values-en-rGB/cm_strings.xml
@@ -210,4 +210,11 @@
   <!-- Format string for fingerprint location message -->
   <!-- Fingerprint sensor locations -->
   <!-- SIM status format string -->
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Network speed</string>
+  <string name="show_network_speed_summary">Show network speed in status bar</string>
+  <string name="status_bar_network_traffic_style_disable">Disabled</string>
+  <string name="status_bar_network_traffic_style_up">Outgoing</string>
+  <string name="status_bar_network_traffic_style_down">Incoming</string>
+  <string name="status_bar_network_traffic_style_all">Outgoing and incoming</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-es-rUS/cm_strings.xml b/packages/apps/Settings/res/values-es-rUS/cm_strings.xml
index 620b080b8..1b73eee48 100644
--- a/packages/apps/Settings/res/values-es-rUS/cm_strings.xml
+++ b/packages/apps/Settings/res/values-es-rUS/cm_strings.xml
@@ -732,6 +732,13 @@
   <string name="status_bar_battery_percentage_default">Oculto</string>
   <string name="status_bar_battery_percentage_text_inside">Dentro del icono</string>
   <string name="status_bar_battery_percentage_text_next">Junto al icono</string>
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Velocidad de red</string>
+  <string name="show_network_speed_summary">Mostrar velocidad de red en la barra de estado</string>
+  <string name="status_bar_network_traffic_style_disable">Deshabilitado</string>
+  <string name="status_bar_network_traffic_style_up">Saliente</string>
+  <string name="status_bar_network_traffic_style_down">Entrante</string>
+  <string name="status_bar_network_traffic_style_all">Salentes y entrantes</string>
   <!-- Status bar - icon blacklist -->
   <string name="status_bar_icons_title">Iconos de la barra de estado</string>
   <string name="status_bar_icons_summary">Controlar qué iconos aparecen en la barra de estado</string>
diff --git a/packages/apps/Settings/res/values-es/cm_strings.xml b/packages/apps/Settings/res/values-es/cm_strings.xml
index bf995df75..e8e216d86 100644
--- a/packages/apps/Settings/res/values-es/cm_strings.xml
+++ b/packages/apps/Settings/res/values-es/cm_strings.xml
@@ -1070,4 +1070,11 @@
   <string name="security_settings_fingerprint_sensor_location_right">lado derecho</string>
   <!-- SIM status format string -->
   <string name="sim_status_format_string">Voz: <xliff:g id="voice_status">%1$s</xliff:g> / Datos: <xliff:g id="data_status">%2$s</xliff:g></string>
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Velocidad de red</string>
+  <string name="show_network_speed_summary">Mostrar velocidad de red en la barra de estado</string>
+  <string name="status_bar_network_traffic_style_disable">Deshabilitado</string>
+  <string name="status_bar_network_traffic_style_up">Datos salientes</string>
+  <string name="status_bar_network_traffic_style_down">Datos entrantes</string>
+  <string name="status_bar_network_traffic_style_all">Salentes y entrantes</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-fr-rCA/cm_strings.xml b/packages/apps/Settings/res/values-fr-rCA/cm_strings.xml
index 384471953..769a48b6d 100644
--- a/packages/apps/Settings/res/values-fr-rCA/cm_strings.xml
+++ b/packages/apps/Settings/res/values-fr-rCA/cm_strings.xml
@@ -197,4 +197,11 @@
   <!-- Format string for fingerprint location message -->
   <!-- Fingerprint sensor locations -->
   <!-- SIM status format string -->
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Vitesse du réseau</string>
+  <string name="show_network_speed_summary">Montrer la vitesse du réseau dans la barre d\'état</string>
+  <string name="status_bar_network_traffic_style_disable">Désactiver</string>
+  <string name="status_bar_network_traffic_style_up">Sortant</string>
+  <string name="status_bar_network_traffic_style_down">Entrant</string>
+  <string name="status_bar_network_traffic_style_all">Sortant et entrant</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-fr/cm_strings.xml b/packages/apps/Settings/res/values-fr/cm_strings.xml
index ea544bffb..3e152588d 100644
--- a/packages/apps/Settings/res/values-fr/cm_strings.xml
+++ b/packages/apps/Settings/res/values-fr/cm_strings.xml
@@ -1065,4 +1065,11 @@ Vous êtes maintenant à <xliff:g id="step_count">%1$d</xliff:g> étapes de l\'a
   <string name="security_settings_fingerprint_sensor_location_left">le côté gauche</string>
   <string name="security_settings_fingerprint_sensor_location_right">le côté droit</string>
   <!-- SIM status format string -->
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Vitesse du réseau</string>
+  <string name="show_network_speed_summary">Montrer la vitesse du réseau dans la barre d\'état</string>
+  <string name="status_bar_network_traffic_style_disable">Désactiver</string>
+  <string name="status_bar_network_traffic_style_up">Sortant</string>
+  <string name="status_bar_network_traffic_style_down">Entrant</string>
+  <string name="status_bar_network_traffic_style_all">Sortant et entrant</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-hu/cm_strings.xml b/packages/apps/Settings/res/values-hu/cm_strings.xml
index 382915e8f..e09500e49 100644
--- a/packages/apps/Settings/res/values-hu/cm_strings.xml
+++ b/packages/apps/Settings/res/values-hu/cm_strings.xml
@@ -1094,4 +1094,11 @@
   <string name="security_settings_fingerprint_sensor_location_right">jobb oldalán</string>
   <!-- SIM status format string -->
   <string name="sim_status_format_string">Hang: <xliff:g id="voice_status">%1$s</xliff:g> / Adat: <xliff:g id="data_status">%2$s</xliff:g></string>
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Hálózat sebessége</string>
+  <string name="show_network_speed_summary">Hálózati sebesség kijelzése az állapotsávon</string>
+  <string name="status_bar_network_traffic_style_disable">Kikapcsolva</string>
+  <string name="status_bar_network_traffic_style_up">Kimenő</string>
+  <string name="status_bar_network_traffic_style_down">Bejövő</string>
+  <string name="status_bar_network_traffic_style_all">Kimenő és bejövő</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-in/cm_strings.xml b/packages/apps/Settings/res/values-in/cm_strings.xml
index 46cdd7c05..7b574a9cc 100644
--- a/packages/apps/Settings/res/values-in/cm_strings.xml
+++ b/packages/apps/Settings/res/values-in/cm_strings.xml
@@ -1093,4 +1093,11 @@
   <string name="security_settings_fingerprint_sensor_location_right">sisi kanan</string>
   <!-- SIM status format string -->
   <string name="sim_status_format_string">Suara: <xliff:g id="voice_status">%1$s</xliff:g> / Data: <xliff:g id="data_status">%2$s</xliff:g></string>
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Kecepatan jaringan</string>
+  <string name="show_network_speed_summary">Tampilkan kecepatan jaringan di status bar</string>
+  <string name="status_bar_network_traffic_style_disable">Dinonaktifkan</string>
+  <string name="status_bar_network_traffic_style_up">Keluar</string>
+  <string name="status_bar_network_traffic_style_down">Masuk</string>
+  <string name="status_bar_network_traffic_style_all">Keluar dan masuk</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-it/cm_strings.xml b/packages/apps/Settings/res/values-it/cm_strings.xml
index 3b5ceb26d..1da07c0c8 100644
--- a/packages/apps/Settings/res/values-it/cm_strings.xml
+++ b/packages/apps/Settings/res/values-it/cm_strings.xml
@@ -1098,4 +1098,11 @@
   <string name="security_settings_fingerprint_sensor_location_right">destra</string>
   <!-- SIM status format string -->
   <string name="sim_status_format_string">Voce: <xliff:g id="voice_status">%1$s</xliff:g> / Dati: <xliff:g id="data_status">%2$s</xliff:g></string>
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Velocità di rete</string>
+  <string name="show_network_speed_summary">Mostra la velocità di rete nella barra di stato</string>
+  <string name="status_bar_network_traffic_style_disable">Disabilitata</string>
+  <string name="status_bar_network_traffic_style_up">In uscita</string>
+  <string name="status_bar_network_traffic_style_down">In entrata</string>
+  <string name="status_bar_network_traffic_style_all">In uscita e in entrata</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-iw/cm_strings.xml b/packages/apps/Settings/res/values-iw/cm_strings.xml
index ec34f2002..022d989dc 100644
--- a/packages/apps/Settings/res/values-iw/cm_strings.xml
+++ b/packages/apps/Settings/res/values-iw/cm_strings.xml
@@ -1096,4 +1096,11 @@
   <string name="security_settings_fingerprint_sensor_location_right">צד ימין</string>
   <!-- SIM status format string -->
   <string name="sim_status_format_string">קול: <xliff:g id="voice_status">%1$s</xliff:g> / נתונים: <xliff:g id="data_status">%2$s</xliff:g></string>
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">מהירות רשת</string>
+  <string name="show_network_speed_summary">הצג את מהירויות הרשת בשורת המצב</string>
+  <string name="status_bar_network_traffic_style_disable">מנוטרל</string>
+  <string name="status_bar_network_traffic_style_up">יוצאות</string>
+  <string name="status_bar_network_traffic_style_down">נכנסות</string>
+  <string name="status_bar_network_traffic_style_all">יוצאות ונכנסות</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-ja/cm_strings.xml b/packages/apps/Settings/res/values-ja/cm_strings.xml
index aa9490529..fe200961c 100644
--- a/packages/apps/Settings/res/values-ja/cm_strings.xml
+++ b/packages/apps/Settings/res/values-ja/cm_strings.xml
@@ -1096,4 +1096,11 @@
   <string name="security_settings_fingerprint_sensor_location_right">右側面</string>
   <!-- SIM status format string -->
   <string name="sim_status_format_string">音声: <xliff:g id="voice_status">%1$s</xliff:g> / データ: <xliff:g id="data_status">%2$s</xliff:g></string>
+  <string name="show_network_speed_title">ネットワーク速度</string>
+  <string name="show_network_speed_summary">ステータスバーにネットワーク速度を表示する</string>
+  <string name="status_bar_network_traffic_style_disable">無効
+</string>
+  <string name="status_bar_network_traffic_style_up">送信</string>
+  <string name="status_bar_network_traffic_style_down">受信</string>
+  <string name="status_bar_network_traffic_style_all">送信と受信</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-ko/cm_strings.xml b/packages/apps/Settings/res/values-ko/cm_strings.xml
index 6ee3f2a5e..356cc739b 100644
--- a/packages/apps/Settings/res/values-ko/cm_strings.xml
+++ b/packages/apps/Settings/res/values-ko/cm_strings.xml
@@ -1033,4 +1033,11 @@
   <!-- Format string for fingerprint location message -->
   <!-- Fingerprint sensor locations -->
   <!-- SIM status format string -->
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">네트워크 속도</string>
+  <string name="show_network_speed_summary">상단바에 네트워크 속도 표시</string>
+  <string name="status_bar_network_traffic_style_disable">사용안함</string>
+  <string name="status_bar_network_traffic_style_up">발신</string>
+  <string name="status_bar_network_traffic_style_down">수신</string>
+  <string name="status_bar_network_traffic_style_all">발신 및 수신</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-nl/cm_strings.xml b/packages/apps/Settings/res/values-nl/cm_strings.xml
index 627b53159..f364f8fed 100644
--- a/packages/apps/Settings/res/values-nl/cm_strings.xml
+++ b/packages/apps/Settings/res/values-nl/cm_strings.xml
@@ -1096,4 +1096,11 @@
   <string name="security_settings_fingerprint_sensor_location_right">rechterkant</string>
   <!-- SIM status format string -->
   <string name="sim_status_format_string">Spraak: <xliff:g id="voice_status">%1$s</xliff:g> / Gegevens: <xliff:g id="data_status">%2$s</xliff:g></string>
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Netwerksnelheid</string>
+  <string name="show_network_speed_summary">Netwerksnelheid in statusbalk weergeven</string>
+  <string name="status_bar_network_traffic_style_disable">Uitgeschakeld</string>
+  <string name="status_bar_network_traffic_style_up">Uitgaand</string>
+  <string name="status_bar_network_traffic_style_down">Inkomend</string>
+  <string name="status_bar_network_traffic_style_all">Uitgaand en inkomend</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-pl/cm_strings.xml b/packages/apps/Settings/res/values-pl/cm_strings.xml
index 0214bd43d..d38195939 100644
--- a/packages/apps/Settings/res/values-pl/cm_strings.xml
+++ b/packages/apps/Settings/res/values-pl/cm_strings.xml
@@ -1077,4 +1077,11 @@
   <string name="security_settings_fingerprint_sensor_location_left">lewa strona</string>
   <string name="security_settings_fingerprint_sensor_location_right">prawa strona</string>
   <!-- SIM status format string -->
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Prędkość transmisji danych</string>
+  <string name="show_network_speed_summary">Pokaż prędkość transmisji danych na pasku stanu</string>
+  <string name="status_bar_network_traffic_style_disable">Wyłączone</string>
+  <string name="status_bar_network_traffic_style_up">Wysłania</string>
+  <string name="status_bar_network_traffic_style_down">Pobierania</string>
+  <string name="status_bar_network_traffic_style_all">Wysyłania i pobierania</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-pt-rBR/cm_strings.xml b/packages/apps/Settings/res/values-pt-rBR/cm_strings.xml
index 1ff832d27..c7241acb9 100644
--- a/packages/apps/Settings/res/values-pt-rBR/cm_strings.xml
+++ b/packages/apps/Settings/res/values-pt-rBR/cm_strings.xml
@@ -1095,4 +1095,11 @@
   <string name="security_settings_fingerprint_sensor_location_right">lado direito</string>
   <!-- SIM status format string -->
   <string name="sim_status_format_string">Voz: <xliff:g id="voice_status">%1$s</xliff:g> / Dados: <xliff:g id="data_status">%2$s</xliff:g></string>
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Velocidade de rede</string>
+  <string name="show_network_speed_summary">Exibir velocidade de rede na barra de status</string>
+  <string name="status_bar_network_traffic_style_disable">Desativado</string>
+  <string name="status_bar_network_traffic_style_up">Enviados</string>
+  <string name="status_bar_network_traffic_style_down">Recebendo</string>
+  <string name="status_bar_network_traffic_style_all">Enviando e recebendo</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-pt-rPT/cm_strings.xml b/packages/apps/Settings/res/values-pt-rPT/cm_strings.xml
index 5cbc7c73d..4e811f824 100644
--- a/packages/apps/Settings/res/values-pt-rPT/cm_strings.xml
+++ b/packages/apps/Settings/res/values-pt-rPT/cm_strings.xml
@@ -1098,4 +1098,6 @@
   <string name="security_settings_fingerprint_sensor_location_right">lateral direita</string>
   <!-- SIM status format string -->
   <string name="sim_status_format_string">Voz: <xliff:g id="voice_status">%1$s</xliff:g> / Dados: <xliff:g id="data_status">%2$s</xliff:g></string>
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Velocidade de rede</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-ru/cm_strings.xml b/packages/apps/Settings/res/values-ru/cm_strings.xml
index bb06f8dc5..8c321dbb5 100644
--- a/packages/apps/Settings/res/values-ru/cm_strings.xml
+++ b/packages/apps/Settings/res/values-ru/cm_strings.xml
@@ -1100,4 +1100,11 @@
   <string name="security_settings_fingerprint_sensor_location_right">правой грани</string>
   <!-- SIM status format string -->
   <string name="sim_status_format_string">Голосовая связь: <xliff:g id="voice_status">%1$s</xliff:g> / Передача данных: <xliff:g id="data_status">%2$s</xliff:g></string>
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Скорость сети</string>
+  <string name="show_network_speed_summary">Показывать скорость передачи данных в строке состояния</string>
+  <string name="status_bar_network_traffic_style_disable">Отключено</string>
+  <string name="status_bar_network_traffic_style_up">Исходящий</string>
+  <string name="status_bar_network_traffic_style_down">Входящий</string>
+  <string name="status_bar_network_traffic_style_all">Исходящий и входящий</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-sl/cm_strings.xml b/packages/apps/Settings/res/values-sl/cm_strings.xml
index c840954df..e16032754 100644
--- a/packages/apps/Settings/res/values-sl/cm_strings.xml
+++ b/packages/apps/Settings/res/values-sl/cm_strings.xml
@@ -1109,4 +1109,11 @@
   <string name="security_settings_fingerprint_sensor_location_right">desni strani</string>
   <!-- SIM status format string -->
   <string name="sim_status_format_string">Glas: <xliff:g id="voice_status">%1$s</xliff:g> / Podatki: <xliff:g id="data_status">%2$s</xliff:g></string>
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Hitrost omrežja</string>
+  <string name="show_network_speed_summary">Prikaži omrežno hitrost v vrstici stanja</string>
+  <string name="status_bar_network_traffic_style_disable">Onemogočen</string>
+  <string name="status_bar_network_traffic_style_up">Odhodni</string>
+  <string name="status_bar_network_traffic_style_down">Dohodni</string>
+  <string name="status_bar_network_traffic_style_all">Odhodni in prejeti</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-sr/cm_strings.xml b/packages/apps/Settings/res/values-sr/cm_strings.xml
index a96daaf27..8494e3da4 100644
--- a/packages/apps/Settings/res/values-sr/cm_strings.xml
+++ b/packages/apps/Settings/res/values-sr/cm_strings.xml
@@ -1097,4 +1097,11 @@
   <string name="security_settings_fingerprint_sensor_location_right">на десној страни</string>
   <!-- SIM status format string -->
   <string name="sim_status_format_string">Глас: <xliff:g id="voice_status">%1$s</xliff:g> / Податак: <xliff:g id="data_status">%2$s</xliff:g></string>
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Приказ протока података</string>
+  <string name="show_network_speed_summary">Приказ брзине протока података у статусној траци</string>
+  <string name="status_bar_network_traffic_style_disable">Онемогућено</string>
+  <string name="status_bar_network_traffic_style_up">Одлазни</string>
+  <string name="status_bar_network_traffic_style_down">Долазни</string>
+  <string name="status_bar_network_traffic_style_all">Одлазни и долазни</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-sv/cm_strings.xml b/packages/apps/Settings/res/values-sv/cm_strings.xml
index 6156bc900..0050b4143 100644
--- a/packages/apps/Settings/res/values-sv/cm_strings.xml
+++ b/packages/apps/Settings/res/values-sv/cm_strings.xml
@@ -1028,4 +1028,11 @@
   <!-- Format string for fingerprint location message -->
   <!-- Fingerprint sensor locations -->
   <!-- SIM status format string -->
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Nätverkshastighet</string>
+  <string name="show_network_speed_summary">Visa nätverkshastighet i statusfältet</string>
+  <string name="status_bar_network_traffic_style_disable">Inaktiverad</string>
+  <string name="status_bar_network_traffic_style_up">Utgående</string>
+  <string name="status_bar_network_traffic_style_down">Inkommande</string>
+  <string name="status_bar_network_traffic_style_all">Utgående och inkommande</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-tr/cm_strings.xml b/packages/apps/Settings/res/values-tr/cm_strings.xml
index 32187be41..978055b3b 100644
--- a/packages/apps/Settings/res/values-tr/cm_strings.xml
+++ b/packages/apps/Settings/res/values-tr/cm_strings.xml
@@ -732,6 +732,13 @@
   <string name="status_bar_battery_percentage_default">Gizli</string>
   <string name="status_bar_battery_percentage_text_inside">Simge içinde</string>
   <string name="status_bar_battery_percentage_text_next">Simge yanında</string>
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Ağ hızı</string>
+  <string name="show_network_speed_summary">Ağ hızını durum çubuğunda göster</string>
+  <string name="status_bar_network_traffic_style_disable">Devre Dışı</string>
+  <string name="status_bar_network_traffic_style_up">Giden</string>
+  <string name="status_bar_network_traffic_style_down">Gelen</string>
+  <string name="status_bar_network_traffic_style_all">Giden ve gelen</string>
   <!-- Status bar - icon blacklist -->
   <string name="status_bar_icons_title">Durum çubuğu simgeleri</string>
   <string name="status_bar_icons_summary">Gösterilen durum çubuğu simgelerini kontrol et</string>
diff --git a/packages/apps/Settings/res/values-uk/cm_strings.xml b/packages/apps/Settings/res/values-uk/cm_strings.xml
index 1425c6968..65158040f 100644
--- a/packages/apps/Settings/res/values-uk/cm_strings.xml
+++ b/packages/apps/Settings/res/values-uk/cm_strings.xml
@@ -944,4 +944,11 @@
   <!-- Format string for fingerprint location message -->
   <!-- Fingerprint sensor locations -->
   <!-- SIM status format string -->
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Швидкість мережі</string>
+  <string name="show_network_speed_summary">Показувати швидкість з\'єднання у рядку стану</string>
+  <string name="status_bar_network_traffic_style_disable">Вимкнено</string>
+  <string name="status_bar_network_traffic_style_up">Вихідний</string>
+  <string name="status_bar_network_traffic_style_down">Вхідний</string>
+  <string name="status_bar_network_traffic_style_all">Вихідний та вхідний</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-vi/cm_strings.xml b/packages/apps/Settings/res/values-vi/cm_strings.xml
index 305468285..90b97adda 100644
--- a/packages/apps/Settings/res/values-vi/cm_strings.xml
+++ b/packages/apps/Settings/res/values-vi/cm_strings.xml
@@ -1042,4 +1042,11 @@
   <!-- Format string for fingerprint location message -->
   <!-- Fingerprint sensor locations -->
   <!-- SIM status format string -->
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">Tốc độ mạng</string>
+  <string name="show_network_speed_summary">Hiển thị tốc độ mạng trên thanh trạng thái</string>
+  <string name="status_bar_network_traffic_style_disable">Đã tắt</string>
+  <string name="status_bar_network_traffic_style_up">Cuộc gọi đi</string>
+  <string name="status_bar_network_traffic_style_down">Cuộc gọi đến</string>
+  <string name="status_bar_network_traffic_style_all">Đi và đến</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-zh-rCN/cm_strings.xml b/packages/apps/Settings/res/values-zh-rCN/cm_strings.xml
index 3ff7ec872..7d04c6762 100644
--- a/packages/apps/Settings/res/values-zh-rCN/cm_strings.xml
+++ b/packages/apps/Settings/res/values-zh-rCN/cm_strings.xml
@@ -1097,4 +1097,11 @@
   <string name="security_settings_fingerprint_sensor_location_right">右侧</string>
   <!-- SIM status format string -->
   <string name="sim_status_format_string">语音：<xliff:g id="voice_status">%1$s</xliff:g> / 数据：<xliff:g id="data_status">%2$s</xliff:g></string>
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">状态栏网速</string>
+  <string name="show_network_speed_summary">在状态栏显示当前网速</string>
+  <string name="status_bar_network_traffic_style_disable">禁用</string>
+  <string name="status_bar_network_traffic_style_up">显示上传速度</string>
+  <string name="status_bar_network_traffic_style_down">显示下载速度</string>
+  <string name="status_bar_network_traffic_style_all">显示上传和下载速度</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-zh-rHK/cm_strings.xml b/packages/apps/Settings/res/values-zh-rHK/cm_strings.xml
index ed0a8dd64..811fffc4d 100644
--- a/packages/apps/Settings/res/values-zh-rHK/cm_strings.xml
+++ b/packages/apps/Settings/res/values-zh-rHK/cm_strings.xml
@@ -837,4 +837,11 @@
   <!-- Format string for fingerprint location message -->
   <!-- Fingerprint sensor locations -->
   <!-- SIM status format string -->
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">狀態列網速</string>
+  <string name="show_network_speed_summary">在狀態列顯示即時網速</string>
+  <string name="status_bar_network_traffic_style_disable">已停用</string>
+  <string name="status_bar_network_traffic_style_up">上載</string>
+  <string name="status_bar_network_traffic_style_down">下載</string>
+  <string name="status_bar_network_traffic_style_all">上載及下載</string>
 </resources>
diff --git a/packages/apps/Settings/res/values-zh-rTW/cm_strings.xml b/packages/apps/Settings/res/values-zh-rTW/cm_strings.xml
index b327c76ef..410136dc1 100644
--- a/packages/apps/Settings/res/values-zh-rTW/cm_strings.xml
+++ b/packages/apps/Settings/res/values-zh-rTW/cm_strings.xml
@@ -1097,4 +1097,11 @@
   <string name="security_settings_fingerprint_sensor_location_right">右側</string>
   <!-- SIM status format string -->
   <string name="sim_status_format_string">語音︰ <xliff:g id="voice_status">%1$s </xliff:g> / 資料︰ <xliff:g id="data_status">%2$s </xliff:g></string>
+  <!-- Show network speed -->
+  <string name="show_network_speed_title">狀態欄網速</string>
+  <string name="show_network_speed_summary">在狀態欄顯示當前網速</string>
+  <string name="status_bar_network_traffic_style_disable">停用</string>
+  <string name="status_bar_network_traffic_style_up">顯示上傳速度</string>
+  <string name="status_bar_network_traffic_style_down">顯示下載速度</string>
+  <string name="status_bar_network_traffic_style_all">顯示上傳和下載速度</string>
 </resources>
diff --git a/packages/apps/Settings/res/values/cm_arrays.xml b/packages/apps/Settings/res/values/cm_arrays.xml
index 57375d9d7..fbd065c61 100644
--- a/packages/apps/Settings/res/values/cm_arrays.xml
+++ b/packages/apps/Settings/res/values/cm_arrays.xml
@@ -610,4 +610,19 @@
         <item>900000</item>
         <item>1800000</item>
     </string-array>
+
+    <!-- show network traffic -->
+    <string-array name="entries_status_bar_network_traffic_style" translatable="false">
+        <item>@string/status_bar_network_traffic_style_disable</item>
+        <item>@string/status_bar_network_traffic_style_up</item>
+        <item>@string/status_bar_network_traffic_style_down</item>
+        <item>@string/status_bar_network_traffic_style_all</item>
+    </string-array>
+
+    <string-array name="values_status_bar_network_traffic_style" translatable="false">
+        <item>0</item>
+        <item>1</item>
+        <item>2</item>
+        <item>3</item>
+    </string-array>
 </resources>
diff --git a/packages/apps/Settings/res/values/cm_strings.xml b/packages/apps/Settings/res/values/cm_strings.xml
index f717cee0f..be67f6fdc 100644
--- a/packages/apps/Settings/res/values/cm_strings.xml
+++ b/packages/apps/Settings/res/values/cm_strings.xml
@@ -1284,4 +1284,12 @@
 
     <!-- SIM status format string -->
     <string name="sim_status_format_string">Voice: <xliff:g id="voice_status">%1$s</xliff:g> / Data: <xliff:g id="data_status">%2$s</xliff:g></string>
+
+    <!-- Show network speed -->
+    <string name="show_network_speed_title">Network speed</string>
+    <string name="show_network_speed_summary">Show network speed in status bar</string>
+    <string name="status_bar_network_traffic_style_disable">Disabled</string>
+    <string name="status_bar_network_traffic_style_up">Outgoing</string>
+    <string name="status_bar_network_traffic_style_down">Incoming</string>
+    <string name="status_bar_network_traffic_style_all">Outgoing and incoming</string>
 </resources>
diff --git a/packages/apps/Settings/res/xml/status_bar_settings.xml b/packages/apps/Settings/res/xml/status_bar_settings.xml
index 92a96ce0d..7c94d396b 100644
--- a/packages/apps/Settings/res/xml/status_bar_settings.xml
+++ b/packages/apps/Settings/res/xml/status_bar_settings.xml
@@ -42,6 +42,14 @@
         android:entries="@array/status_bar_am_pm_entries"
         android:entryValues="@array/status_bar_am_pm_values" />
 
+     <ListPreference
+        android:key="status_bar_network_traffic_style"
+        android:dialogTitle="@string/show_network_speed_title"
+        android:title="@string/show_network_speed_title"
+        android:entries="@array/entries_status_bar_network_traffic_style"
+        android:entryValues="@array/values_status_bar_network_traffic_style"
+        android:defaultValue="3" />
+
     <ListPreference
         android:key="status_bar_battery_style"
         android:title="@string/status_bar_battery_style_title"
diff --git a/packages/apps/Settings/src/com/android/settings/cyanogenmod/StatusBarSettings.java b/packages/apps/Settings/src/com/android/settings/cyanogenmod/StatusBarSettings.java
index 4b37558aa..8137bb902 100644
--- a/packages/apps/Settings/src/com/android/settings/cyanogenmod/StatusBarSettings.java
+++ b/packages/apps/Settings/src/com/android/settings/cyanogenmod/StatusBarSettings.java
@@ -50,6 +50,7 @@ public class StatusBarSettings extends SettingsPreferenceFragment
 
     private static final String STATUS_BAR_CLOCK_STYLE = "status_bar_clock";
     private static final String STATUS_BAR_AM_PM = "status_bar_am_pm";
+    private static final String STATUS_BAR_NETWORK_TRAFFIC_STYLE = "status_bar_network_traffic_style";
     private static final String STATUS_BAR_BATTERY_STYLE = "status_bar_battery_style";
     private static final String STATUS_BAR_SHOW_BATTERY_PERCENT = "status_bar_show_battery_percent";
     private static final String STATUS_BAR_QUICK_QS_PULLDOWN = "qs_quick_pulldown";
@@ -59,6 +60,7 @@ public class StatusBarSettings extends SettingsPreferenceFragment
 
     private ListPreference mStatusBarClock;
     private ListPreference mStatusBarAmPm;
+    private ListPreference mStatusBarNetworkTraffic;
     private ListPreference mStatusBarBattery;
     private ListPreference mStatusBarBatteryShowPercent;
     private ListPreference mQuickPulldown;
@@ -72,6 +74,7 @@ public class StatusBarSettings extends SettingsPreferenceFragment
 
         mStatusBarClock = (ListPreference) findPreference(STATUS_BAR_CLOCK_STYLE);
         mStatusBarAmPm = (ListPreference) findPreference(STATUS_BAR_AM_PM);
+        mStatusBarNetworkTraffic = (ListPreference) findPreference(STATUS_BAR_NETWORK_TRAFFIC_STYLE);
         mStatusBarBattery = (ListPreference) findPreference(STATUS_BAR_BATTERY_STYLE);
         mStatusBarBatteryShowPercent =
                 (ListPreference) findPreference(STATUS_BAR_SHOW_BATTERY_PERCENT);
@@ -94,6 +97,12 @@ public class StatusBarSettings extends SettingsPreferenceFragment
             mStatusBarAmPm.setOnPreferenceChangeListener(this);
         }
 
+        int networkTrafficStyle = CMSettings.System.getInt(resolver,
+                CMSettings.System.STATUS_BAR_NETWORK_TRAFFIC_STYLE, 3);
+        mStatusBarNetworkTraffic.setValue(String.valueOf(networkTrafficStyle));
+        mStatusBarNetworkTraffic.setSummary(mStatusBarNetworkTraffic.getEntry());
+        mStatusBarNetworkTraffic.setOnPreferenceChangeListener(this);
+
         int batteryStyle = CMSettings.System.getInt(resolver,
                 CMSettings.System.STATUS_BAR_BATTERY_STYLE, 0);
         mStatusBarBattery.setValue(String.valueOf(batteryStyle));
@@ -165,6 +174,13 @@ public class StatusBarSettings extends SettingsPreferenceFragment
             mStatusBarBatteryShowPercent.setSummary(
                     mStatusBarBatteryShowPercent.getEntries()[index]);
             return true;
+        } else if (preference == mStatusBarNetworkTraffic) {
+            int networkTrafficStyle = Integer.valueOf((String) newValue);
+            int index = mStatusBarNetworkTraffic.findIndexOfValue((String) newValue);
+            CMSettings.System.putInt(
+                    resolver, CMSettings.System.STATUS_BAR_NETWORK_TRAFFIC_STYLE, networkTrafficStyle);
+            mStatusBarNetworkTraffic.setSummary(mStatusBarNetworkTraffic.getEntries()[index]);
+            return true;
         } else if (preference == mQuickPulldown) {
             int quickPulldown = Integer.valueOf((String) newValue);
             CMSettings.System.putInt(
